flocker/node/agents/test/test_blockdevice.py

```diff
@@ -10,18 +10,401 @@
 
 from zope.interface.verify import verifyObject
 
+from twisted.python.runtime import platform
+from twisted.python.filepath import FilePath
+from twisted.trial.unittest import SynchronousTestCase, SkipTest
+
 from ..blockdevice import (
-    LoopbackBlockDeviceAPI, IBlockDeviceAPI,
+    BlockDeviceDeployer, LoopbackBlockDeviceAPI, IBlockDeviceAPI,
     BlockDeviceVolume, UnknownVolume, AlreadyAttachedVolume,
-    _losetup_list_parse, _losetup_list, UnattachedVolume
+    CreateBlockDeviceDataset, UnattachedVolume,
+    _losetup_list_parse, _losetup_list,
 )
 
-from twisted.python.filepath import FilePath
-from twisted.trial.unittest import SynchronousTestCase, SkipTest
+from ... import InParallel, IDeployer, IStateChange
+from ....control import Dataset, Manifestation, Node, NodeState, Deployment
 
-GIBIBYTE = 2**30
+GIBIBYTE = 2 ** 30
 REALISTIC_BLOCKDEVICE_SIZE = 4 * GIBIBYTE
 
+if not platform.isLinux():
+    # The majority of Flocker isn't supported except on Linux - this test
+    # module just happens to run some code that obviously breaks on some other
+    # platforms.  Rather than skipping each test module individually it would
+    # be nice to have some single global solution.  FLOC-1560, FLOC-1205
+    skip = "flocker.node.agents.blockdevice is only supported on Linux"
+
+
+class BlockDeviceDeployerTests(SynchronousTestCase):
```

Comment:
There's a minimal IDeployer testcase factory that does this check already, you should use it.